Cannot get brain of mobs on 1.19.2

### Bug Type

Console Errors

### Description

I'm unable to get the brain of a sheep mob. I'm using this code:
`EntityAI ai = BukkitBrain.getBrain((Mob) entity).getGoalAI();`
Server version:
`This server is running Paper version git-Paper-142 (MC: 1.19.2) (Implementing API version 1.19.2-R0.1-SNAPSHOT) (Git: ef0e5a6)`
I'm using MobChip shaded and relocated into my plugin.

### Expected Behavior

Return a brain rather than an error message.

### Actual Behavior

I received this error message in console:
```
java.lang.NullPointerException: Cannot read the array length because "array" is null
	at java.util.Arrays.stream(Arrays.java:5428) ~[?:?]
	at be.isach.ultracosmetics.shaded.mobchip.abstraction.ChipUtil1_19_R1.fromNMS(ChipUtil1_19_R1.java:1744) ~[UltraCosmetics-2.9-RELEASE.jar:?]
	at be.isach.ultracosmetics.shaded.mobchip.abstraction.ChipUtil1_19_R1.fromNMS(ChipUtil1_19_R1.java:1919) ~[UltraCosmetics-2.9-RELEASE.jar:?]
	at be.isach.ultracosmetics.shaded.mobchip.abstraction.ChipUtil1_19_R1.lambda$getGoals$0(ChipUtil1_19_R1.java:136) ~[UltraCosmetics-2.9-RELEASE.jar:?]
	at java.lang.Iterable.forEach(Iterable.java:75) ~[?:?]
	at be.isach.ultracosmetics.shaded.mobchip.abstraction.ChipUtil1_19_R1.getGoals(ChipUtil1_19_R1.java:136) ~[UltraCosmetics-2.9-RELEASE.jar:?]
	at be.isach.ultracosmetics.shaded.mobchip.bukkit.BukkitAI.updateMap(BukkitAI.java:36) ~[UltraCosmetics-2.9-RELEASE.jar:?]
	at be.isach.ultracosmetics.shaded.mobchip.bukkit.BukkitAI.<init>(BukkitAI.java:31) ~[UltraCosmetics-2.9-RELEASE.jar:?]
	at be.isach.ultracosmetics.shaded.mobchip.bukkit.BukkitBrain.getGoalAI(BukkitBrain.java:57) ~[UltraCosmetics-2.9-RELEASE.jar:?]
	at be.isach.ultracosmetics.cosmetics.pets.Pet.onEquip(Pet.java:88) ~[UltraCosmetics-2.9-RELEASE.jar:?]
```

I get other, different errors from getting the brains of other mobs, should I open different issues for those or post those here as well?

So far it hasn't worked at all for me. I got this error when I tried with a zombie, and a nearly identical one with cat:
```
[09:39:42] [Server thread/ERROR]: Cannot cast net.minecraft.world.entity.monster.EntityZombie to java.lang.Double
[09:39:42] [Server thread/ERROR]: java.base/java.lang.Class.cast(Class.java:3921)
[09:39:42] [Server thread/ERROR]: UltraCosmetics-2.9-RELEASE.jar//be.isach.ultracosmetics.shaded.mobchip.abstraction.ChipUtil1_19_R1.getObject(ChipUtil1_19_R1.java:1722)
[09:39:42] [Server thread/ERROR]: UltraCosmetics-2.9-RELEASE.jar//be.isach.ultracosmetics.shaded.mobchip.abstraction.ChipUtil1_19_R1.getDouble(ChipUtil1_19_R1.java:1703)
[09:39:42] [Server thread/ERROR]: UltraCosmetics-2.9-RELEASE.jar//be.isach.ultracosmetics.shaded.mobchip.abstraction.ChipUtil1_19_R1.fromNMS(ChipUtil1_19_R1.java:1923)
[09:39:42] [Server thread/ERROR]: UltraCosmetics-2.9-RELEASE.jar//be.isach.ultracosmetics.shaded.mobchip.abstraction.ChipUtil1_19_R1.lambda$getGoals$0(ChipUtil1_19_R1.java:136)
[09:39:42] [Server thread/ERROR]: java.base/java.lang.Iterable.forEach(Iterable.java:75)
[09:39:42] [Server thread/ERROR]: UltraCosmetics-2.9-RELEASE.jar//be.isach.ultracosmetics.shaded.mobchip.abstraction.ChipUtil1_19_R1.getGoals(ChipUtil1_19_R1.java:136)
[09:39:42] [Server thread/ERROR]: UltraCosmetics-2.9-RELEASE.jar//be.isach.ultracosmetics.shaded.mobchip.bukkit.BukkitAI.updateMap(BukkitAI.java:36)
[09:39:42] [Server thread/ERROR]: UltraCosmetics-2.9-RELEASE.jar//be.isach.ultracosmetics.shaded.mobchip.bukkit.BukkitAI.<init>(BukkitAI.java:31)
[09:39:42] [Server thread/ERROR]: UltraCosmetics-2.9-RELEASE.jar//be.isach.ultracosmetics.shaded.mobchip.bukkit.BukkitBrain.getGoalAI(BukkitBrain.java:57)
[09:39:42] [Server thread/ERROR]: UltraCosmetics-2.9-RELEASE.jar//be.isach.ultracosmetics.cosmetics.pets.Pet.onEquip(Pet.java:88)
```
Interestingly this one was just a stack trace. But when it continued on to add a custom pathfinder to the brain, I got a similar (real) exception:
```
java.lang.NullPointerException: Cannot invoke "java.lang.Double.doubleValue()" because the return value of "be.isach.ultracosmetics.shaded.mobchip.abstraction.ChipUtil1_19_R1.getObject(net.minecraft.world.entity.ai.goal.PathfinderGoal, String, java.lang.Class)" is null
	at be.isach.ultracosmetics.shaded.mobchip.abstraction.ChipUtil1_19_R1.getDouble(ChipUtil1_19_R1.java:1703) ~[UltraCosmetics-2.9-RELEASE.jar:?]
	at be.isach.ultracosmetics.shaded.mobchip.abstraction.ChipUtil1_19_R1.fromNMS(ChipUtil1_19_R1.java:1923) ~[UltraCosmetics-2.9-RELEASE.jar:?]
	at be.isach.ultracosmetics.shaded.mobchip.abstraction.ChipUtil1_19_R1.lambda$getGoals$0(ChipUtil1_19_R1.java:136) ~[UltraCosmetics-2.9-RELEASE.jar:?]
	at java.lang.Iterable.forEach(Iterable.java:75) ~[?:?]
	at be.isach.ultracosmetics.shaded.mobchip.abstraction.ChipUtil1_19_R1.getGoals(ChipUtil1_19_R1.java:136) ~[UltraCosmetics-2.9-RELEASE.jar:?]
	at be.isach.ultracosmetics.shaded.mobchip.bukkit.BukkitAI.updateMap(BukkitAI.java:36) ~[UltraCosmetics-2.9-RELEASE.jar:?]
	at be.isach.ultracosmetics.shaded.mobchip.bukkit.BukkitAI.<init>(BukkitAI.java:31) ~[UltraCosmetics-2.9-RELEASE.jar:?]
	at be.isach.ultracosmetics.shaded.mobchip.bukkit.BukkitBrain.getGoalAI(BukkitBrain.java:57) ~[UltraCosmetics-2.9-RELEASE.jar:?]
	at be.isach.ultracosmetics.cosmetics.pets.Pet.onEquip(Pet.java:88) ~[UltraCosmetics-2.9-RELEASE.jar:?]
	at be.isach.ultracosmetics.cosmetics.Cosmetic.equip(Cosmetic.java:84) ~[UltraCosmetics-2.9-RELEASE.jar:?]
	at be.isach.ultracosmetics.cosmetics.type.CosmeticType.equip(CosmeticType.java:93) ~[UltraCosmetics-2.9-RELEASE.jar:?]
	at be.isach.ultracosmetics.cosmetics.type.PetType.equip(PetType.java:90) ~[UltraCosmetics-2.9-RELEASE.jar:?]
        ...
```

I got this error when trying a bat, parrot, and several others:
```
java.lang.ExceptionInInitializerError: null
	at be.isach.ultracosmetics.shaded.mobchip.abstraction.ChipUtil1_19_R1$5.<clinit>(ChipUtil1_19_R1.java:994) ~[UltraCosmetics-2.9-RELEASE.jar:?]
	at be.isach.ultracosmetics.shaded.mobchip.abstraction.ChipUtil1_19_R1.toNMS(ChipUtil1_19_R1.java:1683) ~[UltraCosmetics-2.9-RELEASE.jar:?]
	at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197) ~[?:?]
	at java.util.Spliterators$ArraySpliterator.forEachRemaining(Spliterators.java:992) ~[?:?]
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509) ~[?:?]
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499) ~[?:?]
	at java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150) ~[?:?]
	at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173) ~[?:?]
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234) ~[?:?]
	at java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:596) ~[?:?]
	at be.isach.ultracosmetics.shaded.mobchip.abstraction.ChipUtil1_19_R1.custom(ChipUtil1_19_R1.java:1804) ~[UltraCosmetics-2.9-RELEASE.jar:?]
	at be.isach.ultracosmetics.shaded.mobchip.abstraction.ChipUtil1_19_R1.toNMS(ChipUtil1_19_R1.java:385) ~[UltraCosmetics-2.9-RELEASE.jar:?]
	at be.isach.ultracosmetics.shaded.mobchip.abstraction.ChipUtil1_19_R1.addPathfinder(ChipUtil1_19_R1.java:399) ~[UltraCosmetics-2.9-RELEASE.jar:?]
	at be.isach.ultracosmetics.shaded.mobchip.abstraction.ChipUtil.addPathfinders(ChipUtil.java:53) ~[UltraCosmetics-2.9-RELEASE.jar:?]
	at be.isach.ultracosmetics.shaded.mobchip.bukkit.BukkitAI.updateAI(BukkitAI.java:43) ~[UltraCosmetics-2.9-RELEASE.jar:?]
	at be.isach.ultracosmetics.shaded.mobchip.bukkit.BukkitAI.add(BukkitAI.java:113) ~[UltraCosmetics-2.9-RELEASE.jar:?]
	at be.isach.ultracosmetics.shaded.mobchip.bukkit.BukkitAI.put(BukkitAI.java:70) ~[UltraCosmetics-2.9-RELEASE.jar:?]
	at be.isach.ultracosmetics.cosmetics.pets.Pet.onEquip(Pet.java:90) ~[UltraCosmetics-2.9-RELEASE.jar:?]
        ...
Caused by: java.lang.IllegalArgumentException: Invalid key. Must be [a-z0-9/._-]: 
	at com.google.common.base.Preconditions.checkArgument(Preconditions.java:220) ~[guava-31.0.1-jre.jar:?]
	at org.bukkit.NamespacedKey.<init>(NamespacedKey.java:52) ~[paper-api-1.19.2-R0.1-SNAPSHOT.jar:?]
	at org.bukkit.NamespacedKey.minecraft(NamespacedKey.java:145) ~[paper-api-1.19.2-R0.1-SNAPSHOT.jar:?]
	at be.isach.ultracosmetics.shaded.mobchip.ai.animation.EntityAnimation.<init>(EntityAnimation.java:38) ~[UltraCosmetics-2.9-RELEASE.jar:?]
	at be.isach.ultracosmetics.shaded.mobchip.ai.animation.EntityAnimation.<clinit>(EntityAnimation.java:16) ~[UltraCosmetics-2.9-RELEASE.jar:?]
	... 48 more
```

--- a/1_13_R1/src/main/java/me/gamercoder215/mobchip/abstraction/ChipUtil1_13_R1.java
+++ b/1_13_R1/src/main/java/me/gamercoder215/mobchip/abstraction/ChipUtil1_13_R1.java
@@ -1398,7 +1398,7 @@ public final class ChipUtil1_13_R1 implements ChipUtil {
                     Field f = clazz.getDeclaredField(name);
                     f.setAccessible(true);
                     return cast.cast(f.get(o));
-                } catch (NoSuchFieldException e) {
+                } catch (NoSuchFieldException | ClassCastException e) {
                     if (PathfinderGoal.class.isAssignableFrom(clazz.getSuperclass())) clazz = (Class<? extends PathfinderGoal>) clazz.getSuperclass();
                     else break;
                 }
@@ -1417,7 +1417,10 @@ public final class ChipUtil1_13_R1 implements ChipUtil {
 
     private static BlockPosition toNMS(Location l) { return new BlockPosition(l.getX(), l.getY(), l.getZ()); }
 
-    private static List<ItemStack> fromNMS(RecipeItemStack in) { return Arrays.stream(in.choices).map(CraftItemStack::asBukkitCopy).collect(Collectors.toList()); }
+    private static List<ItemStack> fromNMS(RecipeItemStack in) {
+        in.buildChoices();
+        return Arrays.stream(in.choices).map(CraftItemStack::asBukkitCopy).collect(Collectors.toList());
+    }
 
     private static Sound fromNMS(SoundEffect s) {
         try {

--- a/1_13_R2/src/main/java/me/gamercoder215/mobchip/abstraction/ChipUtil1_13_R2.java
+++ b/1_13_R2/src/main/java/me/gamercoder215/mobchip/abstraction/ChipUtil1_13_R2.java
@@ -1384,7 +1384,7 @@ public class ChipUtil1_13_R2 implements ChipUtil {
                     Field f = clazz.getDeclaredField(name);
                     f.setAccessible(true);
                     return cast.cast(f.get(o));
-                } catch (NoSuchFieldException e) {
+                } catch (NoSuchFieldException | ClassCastException e) {
                     if (PathfinderGoal.class.isAssignableFrom(clazz.getSuperclass())) clazz = (Class<? extends PathfinderGoal>) clazz.getSuperclass();
                     else break;
                 }
@@ -1405,7 +1405,10 @@ public class ChipUtil1_13_R2 implements ChipUtil {
 
     private static BlockPosition toNMS(Location l) { return new BlockPosition(l.getX(), l.getY(), l.getZ()); }
 
-    private static List<ItemStack> fromNMS(RecipeItemStack in) { return Arrays.stream(in.choices).map(CraftItemStack::asBukkitCopy).collect(Collectors.toList()); }
+    private static List<ItemStack> fromNMS(RecipeItemStack in) {
+        in.buildChoices();
+        return Arrays.stream(in.choices).map(CraftItemStack::asBukkitCopy).collect(Collectors.toList());
+    }
 
     private static Sound fromNMS(SoundEffect s) {
         try {

--- a/1_14_R1/src/main/java/me/gamercoder215/mobchip/abstraction/ChipUtil1_14_R1.java
+++ b/1_14_R1/src/main/java/me/gamercoder215/mobchip/abstraction/ChipUtil1_14_R1.java
@@ -1695,7 +1695,7 @@ public class ChipUtil1_14_R1 implements ChipUtil {
                     Field f = clazz.getDeclaredField(name);
                     f.setAccessible(true);
                     return cast.cast(f.get(o));
-                } catch (NoSuchFieldException e) {
+                } catch (NoSuchFieldException | ClassCastException e) {
                     if (PathfinderGoal.class.isAssignableFrom(clazz.getSuperclass())) clazz = (Class<? extends PathfinderGoal>) clazz.getSuperclass();
                     else break;
                 }
@@ -1716,7 +1716,10 @@ public class ChipUtil1_14_R1 implements ChipUtil {
 
     private static BlockPosition toNMS(Location l) { return new BlockPosition(l.getX(), l.getY(), l.getZ()); }
 
-    private static List<ItemStack> fromNMS(RecipeItemStack in) { return Arrays.stream(in.choices).map(CraftItemStack::asBukkitCopy).collect(Collectors.toList()); }
+    private static List<ItemStack> fromNMS(RecipeItemStack in) {
+        in.buildChoices();
+        return Arrays.stream(in.choices).map(CraftItemStack::asBukkitCopy).collect(Collectors.toList());
+    }
 
     private static Sound fromNMS(SoundEffect s) {
         try {

--- a/1_15_R1/src/main/java/me/gamercoder215/mobchip/abstraction/ChipUtil1_15_R1.java
+++ b/1_15_R1/src/main/java/me/gamercoder215/mobchip/abstraction/ChipUtil1_15_R1.java
@@ -1688,7 +1688,7 @@ public class ChipUtil1_15_R1 implements ChipUtil {
                     Field f = clazz.getDeclaredField(name);
                     f.setAccessible(true);
                     return cast.cast(f.get(o));
-                } catch (NoSuchFieldException e) {
+                } catch (NoSuchFieldException | ClassCastException e) {
                     if (PathfinderGoal.class.isAssignableFrom(clazz.getSuperclass())) clazz = (Class<? extends PathfinderGoal>) clazz.getSuperclass();
                     else break;
                 }
@@ -1709,7 +1709,10 @@ public class ChipUtil1_15_R1 implements ChipUtil {
 
     private static BlockPosition toNMS(Location l) { return new BlockPosition(l.getX(), l.getY(), l.getZ()); }
 
-    private static List<ItemStack> fromNMS(RecipeItemStack in) { return Arrays.stream(in.choices).map(CraftItemStack::asBukkitCopy).collect(Collectors.toList()); }
+    private static List<ItemStack> fromNMS(RecipeItemStack in) {
+        in.buildChoices();
+        return Arrays.stream(in.choices).map(CraftItemStack::asBukkitCopy).collect(Collectors.toList());
+    }
 
     private static Sound fromNMS(SoundEffect s) {
         try {

--- a/1_16_R1/src/main/java/me/gamercoder215/mobchip/abstraction/ChipUtil1_16_R1.java
+++ b/1_16_R1/src/main/java/me/gamercoder215/mobchip/abstraction/ChipUtil1_16_R1.java
@@ -1716,7 +1716,7 @@ public class ChipUtil1_16_R1 implements ChipUtil {
                     Field f = clazz.getDeclaredField(name);
                     f.setAccessible(true);
                     return cast.cast(f.get(o));
-                } catch (NoSuchFieldException e) {
+                } catch (NoSuchFieldException | ClassCastException e) {
                     if (PathfinderGoal.class.isAssignableFrom(clazz.getSuperclass())) clazz = (Class<? extends PathfinderGoal>) clazz.getSuperclass();
                     else break;
                 }
@@ -1737,7 +1737,10 @@ public class ChipUtil1_16_R1 implements ChipUtil {
 
     private static BlockPosition toNMS(Location l) { return new BlockPosition(l.getX(), l.getY(), l.getZ()); }
 
-    private static List<ItemStack> fromNMS(RecipeItemStack in) { return Arrays.stream(in.choices).map(CraftItemStack::asBukkitCopy).collect(Collectors.toList()); }
+    private static List<ItemStack> fromNMS(RecipeItemStack in) {
+        in.buildChoices();
+        return Arrays.stream(in.choices).map(CraftItemStack::asBukkitCopy).collect(Collectors.toList());
+    }
 
     private static Sound fromNMS(SoundEffect s) {
         try {

--- a/1_16_R2/src/main/java/me/gamercoder215/mobchip/abstraction/ChipUtil1_16_R2.java
+++ b/1_16_R2/src/main/java/me/gamercoder215/mobchip/abstraction/ChipUtil1_16_R2.java
@@ -1715,7 +1715,7 @@ public class ChipUtil1_16_R2 implements ChipUtil {
                     Field f = clazz.getDeclaredField(name);
                     f.setAccessible(true);
                     return cast.cast(f.get(o));
-                } catch (NoSuchFieldException e) {
+                } catch (NoSuchFieldException | ClassCastException e) {
                     if (PathfinderGoal.class.isAssignableFrom(clazz.getSuperclass())) clazz = (Class<? extends PathfinderGoal>) clazz.getSuperclass();
                     else break;
                 }
@@ -1736,7 +1736,10 @@ public class ChipUtil1_16_R2 implements ChipUtil {
 
     private static BlockPosition toNMS(Location l) { return new BlockPosition(l.getX(), l.getY(), l.getZ()); }
 
-    private static List<ItemStack> fromNMS(RecipeItemStack in) { return Arrays.stream(in.choices).map(CraftItemStack::asBukkitCopy).collect(Collectors.toList()); }
+    private static List<ItemStack> fromNMS(RecipeItemStack in) {
+        in.buildChoices();
+        return Arrays.stream(in.choices).map(CraftItemStack::asBukkitCopy).collect(Collectors.toList());
+    }
 
     private static Sound fromNMS(SoundEffect s) {
         try {

--- a/1_16_R3/src/main/java/me/gamercoder215/mobchip/abstraction/ChipUtil1_16_R3.java
+++ b/1_16_R3/src/main/java/me/gamercoder215/mobchip/abstraction/ChipUtil1_16_R3.java
@@ -1715,7 +1715,7 @@ public class ChipUtil1_16_R3 implements ChipUtil {
                     Field f = clazz.getDeclaredField(name);
                     f.setAccessible(true);
                     return cast.cast(f.get(o));
-                } catch (NoSuchFieldException e) {
+                } catch (NoSuchFieldException | ClassCastException e) {
                     if (PathfinderGoal.class.isAssignableFrom(clazz.getSuperclass())) clazz = (Class<? extends PathfinderGoal>) clazz.getSuperclass();
                     else break;
                 }
@@ -1736,7 +1736,10 @@ public class ChipUtil1_16_R3 implements ChipUtil {
 
     private static BlockPosition toNMS(Location l) { return new BlockPosition(l.getX(), l.getY(), l.getZ()); }
 
-    private static List<ItemStack> fromNMS(RecipeItemStack in) { return Arrays.stream(in.choices).map(CraftItemStack::asBukkitCopy).collect(Collectors.toList()); }
+    private static List<ItemStack> fromNMS(RecipeItemStack in) {
+        in.buildChoices();
+        return Arrays.stream(in.choices).map(CraftItemStack::asBukkitCopy).collect(Collectors.toList());
+    }
 
     private static Sound fromNMS(SoundEffect s) { return CraftSound.getBukkit(s); }
 

--- a/1_17_R1/src/main/java/me/gamercoder215/mobchip/abstraction/ChipUtil1_17_R1.java
+++ b/1_17_R1/src/main/java/me/gamercoder215/mobchip/abstraction/ChipUtil1_17_R1.java
@@ -1696,7 +1696,7 @@ public final class ChipUtil1_17_R1 implements ChipUtil {
                     Field f = clazz.getDeclaredField(name);
                     f.setAccessible(true);
                     return cast.cast(f.get(o));
-                } catch (NoSuchFieldException e) {
+                } catch (NoSuchFieldException | ClassCastException e) {
                     if (PathfinderGoal.class.isAssignableFrom(clazz.getSuperclass())) clazz = (Class<? extends PathfinderGoal>) clazz.getSuperclass();
                     else break;
                 }

--- a/1_18_R1/src/main/java/me/gamercoder215/mobchip/abstraction/ChipUtil1_18_R1.java
+++ b/1_18_R1/src/main/java/me/gamercoder215/mobchip/abstraction/ChipUtil1_18_R1.java
@@ -1695,7 +1695,7 @@ public final class ChipUtil1_18_R1 implements ChipUtil {
                     Field f = clazz.getDeclaredField(name);
                     f.setAccessible(true);
                     return cast.cast(f.get(o));
-                } catch (NoSuchFieldException e) {
+                } catch (NoSuchFieldException | ClassCastException e) {
                     if (Goal.class.isAssignableFrom(clazz.getSuperclass())) clazz = (Class<? extends Goal>) clazz.getSuperclass();
                     else break;
                 }
@@ -1716,7 +1716,7 @@ public final class ChipUtil1_18_R1 implements ChipUtil {
 
     private static BlockPos toNMS(Location l) { return new BlockPos(l.getX(), l.getY(), l.getZ()); }
 
-    private static List<ItemStack> fromNMS(Ingredient in) { return Arrays.stream(in.itemStacks).map(CraftItemStack::asBukkitCopy).collect(Collectors.toList()); }
+    private static List<ItemStack> fromNMS(Ingredient in) { return Arrays.stream(in.getItems()).map(CraftItemStack::asBukkitCopy).collect(Collectors.toList()); }
 
     private static Sound fromNMS(SoundEvent s) { return CraftSound.getBukkit(s); }
 

--- a/1_18_R2/src/main/java/me/gamercoder215/mobchip/abstraction/ChipUtil1_18_R2.java
+++ b/1_18_R2/src/main/java/me/gamercoder215/mobchip/abstraction/ChipUtil1_18_R2.java
@@ -1695,7 +1695,7 @@ public final class ChipUtil1_18_R2 implements ChipUtil {
                     Field f = clazz.getDeclaredField(name);
                     f.setAccessible(true);
                     return cast.cast(f.get(o));
-                } catch (NoSuchFieldException e) {
+                } catch (NoSuchFieldException | ClassCastException e) {
                     if (Goal.class.isAssignableFrom(clazz.getSuperclass())) clazz = (Class<? extends Goal>) clazz.getSuperclass();
                     else break;
                 }
@@ -1716,7 +1716,7 @@ public final class ChipUtil1_18_R2 implements ChipUtil {
 
     private static BlockPos toNMS(Location l) { return new BlockPos(l.getX(), l.getY(), l.getZ()); }
 
-    private static List<ItemStack> fromNMS(Ingredient in) { return Arrays.stream(in.itemStacks).map(CraftItemStack::asBukkitCopy).collect(Collectors.toList()); }
+    private static List<ItemStack> fromNMS(Ingredient in) { return Arrays.stream(in.getItems()).map(CraftItemStack::asBukkitCopy).collect(Collectors.toList()); }
 
     private static Sound fromNMS(SoundEvent s) { return CraftSound.getBukkit(s); }
 

--- a/1_19_R1/src/main/java/me/gamercoder215/mobchip/abstraction/ChipUtil1_19_R1.java
+++ b/1_19_R1/src/main/java/me/gamercoder215/mobchip/abstraction/ChipUtil1_19_R1.java
@@ -1720,7 +1720,7 @@ public final class ChipUtil1_19_R1 implements ChipUtil {
                     Field f = clazz.getDeclaredField(name);
                     f.setAccessible(true);
                     return cast.cast(f.get(o));
-                } catch (NoSuchFieldException e) {
+                } catch (NoSuchFieldException | ClassCastException e) {
                     if (Goal.class.isAssignableFrom(clazz.getSuperclass())) clazz = (Class<? extends Goal>) clazz.getSuperclass();
                     else break;
                 }
@@ -1741,7 +1741,7 @@ public final class ChipUtil1_19_R1 implements ChipUtil {
 
     private static BlockPos toNMS(Location l) { return new BlockPos(l.getX(), l.getY(), l.getZ()); }
 
-    private static List<ItemStack> fromNMS(Ingredient in) { return Arrays.stream(in.itemStacks).map(CraftItemStack::asBukkitCopy).collect(Collectors.toList()); }
+    private static List<ItemStack> fromNMS(Ingredient in) { return Arrays.stream(in.getItems()).map(CraftItemStack::asBukkitCopy).collect(Collectors.toList()); }
 
     private static Sound fromNMS(SoundEvent s) { return CraftSound.getBukkit(s); }
 

--- a/base/src/main/java/me/gamercoder215/mobchip/ai/animation/EntityAnimation.java
+++ b/base/src/main/java/me/gamercoder215/mobchip/ai/animation/EntityAnimation.java
@@ -11,10 +11,6 @@ import org.jetbrains.annotations.Nullable;
 public enum EntityAnimation implements Keyed {
 
     /**
-     * Represents an Empty Animation
-     */
-    EMPTY(""),
-    /**
      * Represents the Damage Animation when taking damage
      */
     DAMAGE("hurt"),

